source/databases-collection.txt

```diff
@@ -0,0 +1,265 @@
+.. _ruby-databases-collections:
+
+=========================
+Databases and Collections
+=========================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: table, row, organize, storage
+
+Overview
+--------
+
+In this guide, you can learn how to use MongoDB databases and
+collections with {+driver-short+}.
+
+MongoDB organizes data into a hierarchy of the following levels:
+
+- **Databases**: The top level of data organization in a MongoDB instance.
+- **Collections**: MongoDB stores documents in collections. They are analogous to tables in relational databases.
+- **Documents**: Contain literal data such as string, numbers, dates, and other embedded documents.
+
+For more information about document field types and structure, see the
+:manual:`Documents </core/document/>` guide in the {+mdb-server+} manual.
+
+.. TODO: Add a diagram here
+
+Access a Database
+-----------------
+
+Access a database by creating a ``Mongo::Client`` instance with the desired 
+database name.
+
+The following example accesses a database named ``test_database``:
+
+.. literalinclude:: /read/databases-collection.rb
+   :language: ruby
+   :dedent:
+   :start-after: start-access-db
+   :end-before: end-access-db
+
+
+Access a Collection
+-------------------
+
+Access a collection by using the ``[]`` method on an instance 
+of your database.
+
+The following example accesses a collection named ``test_collection``:
+
+.. literalinclude:: /read/databases-collection.rb
+   :language: ruby
+   :dedent:
+   :emphasize-lines: 2
+   :start-after: start-access-cl
+   :end-before: end-access-cl
+
+.. tip::
+
+   If the provided collection name does not already exist in the database,
+   MongoDB implicitly creates the collection when you first insert data
+   into it.
+
+Create a Collection
+-------------------
+
+While the Ruby driver for MongoDB does not have a direct ``create_collection`` 
+method, you can use the ``database.command`` method with the create command to 
+create a collection with specific options.
```

Comment:
There actually _is_ a method for creating collections. You call `create` on the collection returned by the `#[]` method, e.g.

```ruby
client = Mongo::Client.new(['127.0.0.1:27017'], database: 'test_database')
database = client.database

database[:example_collection].create(capped: true, size: 1024)
```

Relevant API docs are here: https://www.mongodb.com/docs/ruby-driver/current/api/Mongo/Collection.html#create-instance_method